app/controllers/openstax/api/v1/api_controller.rb

```diff
@@ -27,11 +27,11 @@ class ApiController < ActionController::Base
         before_filter :doorkeeper_authorize!, if: :token_user?
 
         # Except for users logged in via a cookie, we can disable CSRF protection and enable CORS
-        skip_before_filter :verify_authenticity_token, unless: :session_user?
+        skip_before_filter :verify_authenticity_token, unless: :local_session_user?
+        skip_before_filter :authenticate_user!, only: :options
         skip_before_filter :verify_authenticity_token, only: :options
-        before_filter :set_cors_preflight_headers, only: :options
-        before_filter :set_cors_headers
-        after_filter :set_cors_headers
+
+        before_filter  :maybe_set_cors_headers
 
```

Comment:
I took out the `after_filter` here because I didn't see why the headers should be set both `before_` and `after_`.  Maybe there's a reason I'm not aware of though?